packages/relay/src/lib/eth.ts

```diff
@@ -1109,69 +1110,96 @@ export class EthImpl implements Eth {
-          throw predefined.INTERNAL_ERROR();
-        }
+    if (!txSubmitted) {
+      return predefined.INTERNAL_ERROR(e.message.toString());
+    }
 
-        return  EthImpl.prepend0x(Buffer.from(record.ethereumHash).toString('hex'));
-      } catch (e) {
+    await this.mirrorNodeClient.getContractRevertReasonFromTransaction(e, requestId, requestIdPrefix);
 
-        await this.mirrorNodeClient.getContractRevertReasonFromTransaction(e, requestId, requestIdPrefix);
+    this.logger.error(e,
+        `${requestIdPrefix} Failed sendRawTransaction during record retrieval for transaction ${transaction}, returning computed hash`);
+    //Return computed hash if unable to retrieve EthereumHash from record due to error
+    return EthImpl.prepend0x(createHash('keccak256').update(transactionBuffer).digest('hex'));
+  }
 
-        this.logger.error(e,
-          `${requestIdPrefix} Failed sendRawTransaction during record retrieval for transaction ${transaction}, returning computed hash`);
-        //Return computed hash if unable to retrieve EthereumHash from record due to error
-        return EthImpl.prepend0x(createHash('keccak256').update(transactionBuffer).digest('hex'));
-      }
-    } catch (e: any) {
-      this.logger.error(e,
-        `${requestIdPrefix} Failed to successfully submit sendRawTransaction for transaction ${transaction}`);
-      if (e instanceof JsonRpcError) {
-        return e;
+  /**
+   * Submits a transaction to the network for execution.
+   *
+   * @param transaction
+   */
+  async sendRawTransaction(transaction: string, requestId?: string): Promise<string | JsonRpcError> {
+    const requestIdPrefix = formatRequestIdMessage(requestId);
+    if(transaction?.length >= constants.FUNCTION_SELECTOR_CHAR_LENGTH)
+      this.ethExecutionsCounter.labels(EthImpl.ethSendRawTransaction, transaction.substring(0, constants.FUNCTION_SELECTOR_CHAR_LENGTH)).inc();
+
+    const parsedTx = await this.parseRawTxAndPrecheck(transaction, requestId);
+    const transactionBuffer = Buffer.from(EthImpl.prune0x(transaction), 'hex');
+
+    let txSubmitted = false;
+    try {
+      const contractExecuteResponse = await this.hapiService.getSDKClient().submitEthereumTransaction(transactionBuffer, EthImpl.ethSendRawTransaction, requestId);
+      txSubmitted = true;
+      // Wait for the record from the execution.
+      let txId = contractExecuteResponse.transactionId.toString();
+      const formattedId = formatTransactionId(txId);
+      const  record = await this.mirrorNodeClient.repeatedRequest('getContractResult', [formattedId], 3, requestId);
+      if (!record) {
+        this.logger.warn(`${requestIdPrefix} No record retrieved`);
+        const tx = await this.mirrorNodeClient.getTransactionById(txId, 0, requestId);
+        if (tx.transactions?.length) {
+          const result = tx.transactions[0].result;
+          if (result === 'WRONG_NONCE') {
```

Comment:
`'WRONG_NONCE'` should also be a constant

Reply:
instead of declaring a const, I am referencing the name of the enum defined at the response protobuf. using the hashgraph lib
https://github.com/hashgraph/hedera-protobufs/blob/c1de1a199d11e6ace89679af9e4500cb92eb841c/services/response_code.proto#L152

Let me know your thoughts.

Reply:
I nevertheless  created its own constants were we can put the values we use of the transactionEnum response status.